### DISCUSSION TITLE
Live preview production preview

### DISCUSSION THREAD

**Discussion by xSketchy0** [sanity-io/sanity#3727] [Sanity Studio Support] [answered]
Hey,

I'm tinkering around with V3 Studio and I was trying to replicate the live preview for V3. How would I go about adding the production-preview plugin to V3 and setting up the resolveProductionUrl file as it's currently unclear for me what file to import and where to create the resolve file.

Thank you!

**Comment by danieljb** [comment, marked as answer]
Hi, we have a little setup running with Remix and Studio v3 where we use `resolveProductionUrl`. We set this up [following the documentation](https://beta.sanity.io/docs/platform/studio/v2-to-v3/prod-url-resolver) and replacing the deprecated `client` from the context with `getClient`.

The studio configuration looks something like this:
```javascript
<Studio
  config={{
    basePath: "/studio",
    projectId,
    dataset,
    apiVersion,
    plugins: [deskTool()],
    document: {
      productionUrl: resolveProductionUrl,
    }
  }}
/>
```

Make sure `projectId`, `dataset` and `apiVersion` are defined and add a function `resolveProductionUrl`:
```javascript
async function resolveProductionUrl(prev, context) {
  const { getClient, document } = context;

  // generate the preview url according to your setup, e.g.:
  if (document._type == "page") {
    const client = await getClient({ apiVersion });
    const slug = client.fetch(
      `*[_id == $id][0].slug.current`,
      { id: document._id }
    );

    const url = new URL("https://my-site.com");
    url.pathname = `/pages/${slug}/`;
    url.searchParams.set('preview', 'true');

    return url.toString();
  }

  return prev;
}
```

This is working with `dev-preview.20`, haven’t tried `21` yet. The `production-preview` plugin is not required any more.

**Reply by quirm** [reply]
`document` seems to be up-to-date. I'm using it to read the necessary values over calling another request. In your case, you could do the following:
```
if (document._type === 'page' && document.slug?.current) {
  const slug = document.slug.current;
  ...
}
```

_(I wonder if it's the right approach)_
